### DISCUSSION TITLE
Bootstrapping clang/wasm toolchain

### DISCUSSION THREAD

**Discussion by vadimkantorov** [emscripten-core/emscripten#22507] [Ideas]
Curious, has anyone tried/succeeded compiling clang/llvm/wasm/emscripten toolchain itself to make an in-browser C compiler able to produce and execute WASM modules directly in-browser?

I found some hobby attempts at writing C-to-wasm compilers from scratch, but if LLVM/Clang itself could be compiled to WASM, this could open way to many more other compilers-in-browser, even if using full Clang for example program C compilation is overkill.

Probably could be interesting for pedagogical purposes / teaching C and alleviating the server costs :)

**Comment by kripken** [comment]
Yes, see [Emception](https://github.com/jprendes/emception) by @jprendes . That's the best one I'm aware of.

**Reply by vadimkantorov** [reply]
Oh wow. Interesting that it's similar to my busytex in that it's also making a busybox-like executable 

I would hope that bootstrapping llvm becomes part of emscripten proper in some future

Unfortunately, emception demo crashes the mobile Safari
